### DISCUSSION TITLE
Is it possible to use multiple sets in a class that inherits AbstractSeries in v5?

### DISCUSSION THREAD

**Discussion by s-ohnishi** [phalcon/cphalcon#16012] [General]
It's no longer used now, but isn't it possible to load a specific .js and .css first and then another .js and .css like jQueryMobile?
```
<link rel="stylesheet" href="css/mytheme.css">
<link rel="stylesheet" href="http://code.jquery.com/mobile/1.0/jquery.mobile-1.0.min.css" />
<script src="http://code.jquery.com/jquery-1.6.4.min.js"></ script>
<script src="http://code.jquery.com/mobile/1.0/jquery.mobile-1.0.min.js"></ script>
<link rel="stylesheet" href="/css/override.min.css" />
<script src="/js/original.min.js" />
```

**Comment by s-ohnishi** [comment]
I came to think that instead of using only the tag property provided by the controller, I should prepare properties named preTag and postTag (each instance of TagFactory).

**Reply by niden** [reply]
Unless you add the code directly yourself you are not going to be able to do that with the `AbstractSeries` because each of them is of the same "type"

To do 

```
<link rel='stylesheet' .... >
<link rel='stylesheet' .... >
<script ... >
<script ...>
<link rel='stylesheet' .... >
<link rel='stylesheet' .... >
```
you will need to create three objects. One before the `<script...>` one for the `<script...>` and one after.

The same applies if you are using the `Assets` component. You will need the `header` collection but more specifically you will need to do `header_css_pre` (for the first few), then `header_js` then whatever comes after could be `header_css_post`

**Reply by s-ohnishi** [reply]
I would like to deal with it by writing it directly in action.volt in HTML or by preparing multiple TagFactory.
I haven't learned about the `Assets` component yet, so I'll take the time to do so.
Thank you very much.

**Reply by niden** [reply]
You might find it useful to use the `Assets`. Have a look at this simple example:

**Controller**
```php
$assets
    ->collection('header_css_pre')
    ->addCss("css/mytheme.css")
    ->addCss("https://code.jquery.com/mobile/1.0/jquery.mobile-1.0.min.css")
;

$assets
    ->collection('header_js')
    ->addJs("https://code.jquery.com/jquery-1.6.4.min.js")
    ->addJs("https://code.jquery.com/mobile/1.0/jquery.mobile-1.0.min.js")
;

$assets
    ->collection('header_css_post')
    ->addCss("/css/override.min.css")
    ->addCss("/js/original.min.js")
;
```

And then in your template you just need to output each collection where you want it. The `Assets` component is already registered in the `FactoryDefault` DI container

```twig
    {{- assets.outputCss('header_css_pre') -}}
    {{- assets.outputJs('header_js') -}}
    {{- assets.outputCss('header_css_post') -}}
```

**Reply by s-ohnishi** [reply]
Thank you for the sample.
I will also read Crowdin's explanation.
I thought I had done some translation of Assets recently, so I looked it up, but my Japanese translation was changed to English.
It's a little different from the source string, so I think it was changed.
However, I can understand if the source string has been changed, but the part translated into Japanese seems to be in English (after the change?) Is this intentional?
Or is it Crowdin's specification?

**Reply by niden** [reply]
I have seen that happen too with the Greek translation. I am guessing some glitch with the Crowdin system.

From what I know, if a string is the same, it stays like that. If it changes even slightly you have to select and approve a new translation with whatever changes needed.

**Reply by s-ohnishi** [reply]
[the sample](https://crowdin.com/translate/phalcon-documentation/5915/enus-ja?filter=basic&value=0#q=825028)

SOURCE STRING
```This class is a helper class that extends the <0>Phalcon\Assets\Asset</0> class and internally sets the first parameter to <1>js</1>.```

current translation(you wrote 3 days ago)
```This class is a helper class that extends the <a href="api/phalcon_assets#assets-asset" f-id="asset" fo="12">Phalcon\Assets\Asset</0> class and internally sets the first parameter to <1>js</1>.```

I wrote 8 days ago
```このクラスは、 <0>Phalcon\Assets\Asset</0> クラスを拡張し、内部的に最初のパラメータを <1>js</1> に設定するヘルパークラスです。```

In this state, English sentences are likely to be displayed as "Japanese".
And if a translation volunteer finds this, no one will suspect the SOURCE STRING, and will only assume that someone has pasted the wrong sentence (and untranslated English) as a translation.

What should we do?
There are probably many more.

**Reply by niden** [reply]
I really don't know why it did that. I can assure you I did not touch Japanese ;) 

The major work that I have done was to hide the code fragments so that they are not translated as well as some keywords.

I made you proofreader for Japanese and now you can approve translations, which will send them to the main site. If you see duplicate stuff like that, just approve the correct translation and delete the other one (if it is from me).

I will send a message to Crowdin, maybe there is a mishap there.

**Reply by s-ohnishi** [reply]
So, in this case, is the correct SOURCE STRING or the one from 3 days ago?
Somehow, I feel that SOURCE STRING is an old one before correction.

**Reply by niden** [reply]
The source string is correct, I double checked it with the `en` folder (which is really the source)

**Reply by s-ohnishi** [reply]
OK.
I'll fix them one by one when I find them.
